picard/ui/logview.py

```diff
@@ -97,31 +125,159 @@ def restoreWindowState(self, position, size):
         config.Option("persist", "logview_size", QtCore.QSize(560, 400)),
+        config.Option("persist", "logview_verbosity", LogViewCommon.default_verbosity)
     ]
 
+    _log_level_names = OrderedDict([
+        (log.LOG_INFO,      N_('Info')),
+        (log.LOG_WARNING,   N_('Warning')),
+        (log.LOG_ERROR,     N_('Error')),
+        (log.LOG_DEBUG,     N_('Debug')),
+    ])
+
     def __init__(self, parent=None):
         title = _("Log")
         logger = log.main_logger
         LogViewCommon.__init__(self, title, logger, parent=parent)
+        self.verbosity = config.persist['logview_verbosity']
         self.restoreWindowState("logview_position", "logview_size")
+        self.hbox = QtWidgets.QHBoxLayout(self)
+        self.vbox.addLayout(self.hbox)
+
         cb = QtWidgets.QCheckBox(_('Debug mode'), self)
         cb.setChecked(QtCore.QObject.tagger._debug)
         cb.stateChanged.connect(self.toggleDebug)
-        self.vbox.addWidget(cb)
+        self.hbox.addWidget(cb)
+
+        self.verbosity_menu = QtWidgets.QMenu(self)
+        for level, label in self._log_level_names.items():
+            act = QtWidgets.QAction(_(label), self.verbosity_menu)
+            act.setCheckable(True)
+            act.setChecked(level in self.verbosity)
+            act.triggered.connect(partial(self._verbosity_changed, level))
+            self.verbosity_menu.addAction(act)
+
+        self.verbosity_menu_button = QtWidgets.QPushButton(_("Verbosity"))
+        self.verbosity_menu_button.setMenu(self.verbosity_menu)
+        self.hbox.addWidget(self.verbosity_menu_button)
+
+        self.domains_menu_can_update = True
+
+        self.domains_menu = QtWidgets.QMenu(self)
+        self.domains_menu_button = QtWidgets.QPushButton(_("Domains"))
+        self.domains_menu_button.setMenu(self.domains_menu)
+        self.domains_menu.aboutToShow.connect(partial(self.set_domains_menu_can_update, False))
+        self.domains_menu.aboutToHide.connect(partial(self.set_domains_menu_can_update, True))
+
+        self.show_only_domains_menu = QtWidgets.QMenu(self)
+        self.show_only_domains_menu_button = QtWidgets.QPushButton(_("Show Only"))
+        self.show_only_domains_menu_button.setMenu(self.show_only_domains_menu)
+        self.show_only_domains_menu.aboutToShow.connect(partial(self.set_domains_menu_can_update, False))
+        self.show_only_domains_menu.aboutToHide.connect(partial(self.set_domains_menu_can_update, True))
+
+        self.hbox.addWidget(self.domains_menu_button)
+        self.hbox.addWidget(self.show_only_domains_menu_button)
+
+        self.highlight_text = QtWidgets.QLineEdit()
+        self.hbox.addWidget(self.highlight_text)
+
+        self.highlight_button = QtWidgets.QPushButton(_("Highlight"))
+        self.hbox.addWidget(self.highlight_button)
+        self.highlight_button.clicked.connect(self._highlight_do)
+
+        self.clear_log_button = QtWidgets.QPushButton(_("Clear Log"))
```

Comment:
Can we add some tooltips for all these options? It is unclear to a user whether these are destructive or just temp.

Reply:
Yes, of course, this option actually clears the circular buffer. UI tuning will come at the end.